azure/services/managedclusters/spec.go

```diff
@@ -380,429 +386,209 @@ func (s *ManagedClusterSpec) Parameters(ctx context.Context, existing interface{
-		}
-		ps := *existingMC.Properties.ProvisioningState
-		if ps != string(infrav1.Canceled) && ps != string(infrav1.Failed) && ps != string(infrav1.Succeeded) {
-			return nil, azure.WithTransientError(errors.Errorf("Unable to update existing managed cluster in non-terminal state. Managed cluster must be in one of the following provisioning states: Canceled, Failed, or Succeeded. Actual state: %s", ps), 20*time.Second)
-		}
-
-		if managedCluster.Properties.AddonProfiles == nil && existingMC.Properties.AddonProfiles != nil {
-			managedCluster.Properties.AddonProfiles = map[string]*armcontainerservice.ManagedClusterAddonProfile{}
-		}
-		for key, item := range existingMC.Properties.AddonProfiles {
-			if _, ok := managedCluster.Properties.AddonProfiles[key]; !ok {
-				addonProfile := &armcontainerservice.ManagedClusterAddonProfile{
-					Enabled: ptr.To(false),
-				}
-				if item.Config != nil {
-					addonProfile.Config = item.Config
-				}
-				managedCluster.Properties.AddonProfiles[key] = addonProfile
-			}
-			existingMC.Properties.AddonProfiles[key].Identity = nil
-		}
-		// Normalize the LoadBalancerProfile so the diff below doesn't get thrown off by AKS added properties.
-		if managedCluster.Properties.NetworkProfile.LoadBalancerProfile == nil {
-			// If our LoadBalancerProfile generated by the spec is nil, then don't worry about what AKS has added.
-			existingMC.Properties.NetworkProfile.LoadBalancerProfile = nil
-		} else {
-			// If our LoadBalancerProfile generated by the spec is not nil, then remove the effective outbound IPs from
-			// AKS.
-			existingMC.Properties.NetworkProfile.LoadBalancerProfile.EffectiveOutboundIPs = nil
-		}
-
-		// Avoid changing agent pool profiles through AMCP and just use the existing agent pool profiles
-		// AgentPool changes are managed through AMMP.
-		managedCluster.Properties.AgentPoolProfiles = existingMC.Properties.AgentPoolProfiles
-
-		// if the AuthorizedIPRanges is nil in the user-updated spec, but not nil in the existing spec, then
-		// we need to set the AuthorizedIPRanges to empty array ([]*string{}) once so that the Azure API will
-		// update the existing authorized IP ranges to nil.
-		if !isAuthIPRangesNilOrEmpty(existingMC) && isAuthIPRangesNilOrEmpty(managedCluster) {
-			log.V(4).Info("managed cluster spec has nil AuthorizedIPRanges, updating existing authorized IP ranges to an empty list")
-			managedCluster.Properties.APIServerAccessProfile = &armcontainerservice.ManagedClusterAPIServerAccessProfile{
-				AuthorizedIPRanges: []*string{},
-			}
-		}
-
-		diff := computeDiffOfNormalizedClusters(managedCluster, existingMC)
-		if diff == "" {
-			log.V(4).Info("no changes found between user-updated spec and existing spec")
-			return nil, nil
-		}
-		log.V(4).Info("found a diff between the desired spec and the existing managed cluster", "difference", diff)
-	} else {
+	// Only include AgentPoolProfiles during initial cluster creation. Agent pools are managed solely by the
```

Comment:
Yes, this is still necessary for now since AKS requires them on create. And if we leave the agentPoolProfiles in the ManagedCluster spec after it's created, then later updates to the ManagedCluster may override the ManagedClustersAgentPools.